### DISCUSSION TITLE
Difference between array.py and m_array.py

### DISCUSSION THREAD

**Discussion by ttzytt** [drageelr/manim-data-structures#16] [Q&A] [answered]
I cloned the GitHub repository and tried to run a few pieces of examples from the documentation:

```py
class MyScene(Scene):
    def construct(self):
        arr = MArray(self, [1, 2, 3])
        self.play(Create(arr))
        self.wait(1)
```` 

An error just occurred: 

```
│ E:\pythons\3.11\Lib\site-packages\manim_data_structures\array.py:470 in __init__                 │
│                                                                                                  │
│   467 │   │                                                                                      │
│   468 │   │   print("arr ", arr)                                                                 │
│   469 │   │                                                                                      │
│ ❱ 470 │   │   for v in arr:                                                                      │
│   471 │   │   │   self.__append_elem(v, mob_square_args, mob_value_args, mob_index_args)         │
│   472 │                                                                                          │
│   473 │   def update_elem_value(self, index: int, value, mob_value_args: dict = {}) -> Text:     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: 'MyScene' object is not iterable
```

It seems like the package I downloaded from pip is different from what I cloned from GitHub. The parameter of `__init__` function for `m_array.py` (from the Github repo), and `array.py` is different. One of them uses `scene` as a parameter but the other does not. 

After deleting the parameter of `self`, everything worked correctly.

So I'm wondering what caused this inconsistency. Is it because the package from pip is not updated? Or just because I'm not using this correctly?

**Comment by drageelr** [comment, marked as answer]
Hello @ttzytt 
From version 0.1.6 onward, the `Scene` needs to be passed as the first argument to `MArray` and other such classes. The package is published to PyPi using the `PYPI` workflow defined on the `main` branch, so imo the repo's `main` branch is consistent with the latest package published.

Moreover, the file `array.py` was renamed to `m_array.py` in the release 0.1.5 (as mentioned in the changelog). Since you mentioned that these two files differ I think that you haven't updated the main branch for your clone.

Try pulling the latest commits from the `main` branch and let me know if the error still persists.

**Reply by drageelr** [reply]
@ttzytt Just re-read your comment. I think your repo is updated but the pip package is not, kindly update it to at least 0.1.6 and the error would be resolved.

**Comment by ttzytt** [comment]
@drageelr Thanks a lot. Now I can run the code correctly by using `sys.path.append()` to add the path of the cloned repo.  Since the repo is updated, so I directly used it without the pip package.

**Reply by drageelr** [reply]
You can also run `poetry shell` while in the repo's directory and execute `pip install .` to install the package from the repo. However, make sure to invoke manim via poetry shell.
